x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/delayeddatacheck/DatafeedDelayedDataDetector.java

```diff
@@ -134,9 +138,19 @@ private Map<Long, Long> checkCurrentBucketEventCount(long start, long end) {
 
         SearchRequest searchRequest = new SearchRequest(datafeedIndices).source(searchSourceBuilder).indicesOptions(indicesOptions);
         try (ThreadContext.StoredContext ignore = client.threadPool().getThreadContext().stashWithOrigin(ML_ORIGIN)) {
-            SearchResponse response = client.execute(TransportSearchAction.TYPE, searchRequest).actionGet();
+            SearchResponse searchResponse = client.execute(TransportSearchAction.TYPE, searchRequest).actionGet();
             try {
-                List<? extends Histogram.Bucket> buckets = ((Histogram) response.getAggregations().get(DATE_BUCKETS)).getBuckets();
+                Histogram histogram = searchResponse.getAggregations().get(DATE_BUCKETS);
+                if (histogram == null) {
+                    logger.debug(
+                        "Missing [{}] aggregation in the search response, request = [{}], response = [{}]",
```

Comment:
Include the job Id and the fact that this error stems from the delayed data check. 

```suggestion
                        "[{}] Delayed data check failed with missing aggregation in search response", jobId
```